packages/knip/src/typescript/getImportsAndExports.ts

```diff
@@ -387,7 +387,7 @@ const getImportsAndExports = (
 
   const setRefs = (item: SerializableExport | SerializableExportMember) => {
     if (!item.symbol) return;
-    for (const match of sourceFile.text.matchAll(new RegExp(item.identifier.replace(/\$/g, '\\$'), 'g'))) {
+    for (const match of sourceFile.text.matchAll(new RegExp(item.identifier.replace(/[\\^$*+?.()|[\]{}]/g, '\\$&'), 'g'))) {
```

Comment:
This targets (enum) members, perhaps it makes sense to only apply this to members at https://github.com/webpro/knip/blob/main/packages/knip/src/typescript/getImportsAndExports.ts#L43? Ideally we'd create the regex once and reuse it (I should've done that myself here too).

Reply:
@webpro do you mean to make the escaping of special characters already when creating the serializableMember? I've just had a cursory look, and it looks like modifying the `identifier` would have lot's of other effects, but perhaps adding another `regex` property to the serializable members which does identifier->escape->regex and then we can use that pre-created regex here for the `matchAll()`?

Reply:
Yeah ok I had another look, and I can't figure out what exactly you mean. I've moved the Regex into it's own file to be reused, also tried to search for anywhere in the code here a similiar regex was used, but couldn't find anywhere. 
Let me know what you think!